### ISSUE TITLE
Templates edit action

### ISSUE THREAD

**Issue by Ashark** [gmm96/KDE-5-Template-Manager#7]
It seems that it is in development.
Some items are not displayed properly. For example, blank spaces instead of naming actions such as editing icon.
Also, when installing via Dolphin (get new thing button), there is no such action.
Looks like it is not yet published in https://www.pling.com/p/1454391/

**Comment by gmm96** [comment]
As you finally said, it is not published yet in the _Dolphin store_ because I need to make some more tests.
Which blank spaces do you mean? Can you please show me a screenshot pointing to that?

Kdialog (used tool to input from user) has limitations, specially formatting checklist options, I cannot use bold, italic, center or anything like that. My intention was to display the action and the current value associated below, something like this:

![image](https://raw.githubusercontent.com/gmm96/KDE-5-Template-Manager/main/screenshots/edit10.png)

In this case, there's a blank line after _Edit comment_ action, which means it has no value right now. Also after _Save and exit_ action, but this action won't obviously have any value associated. Just in case these are the blank lines you're talking about.

I know it isn't the most usable interface you're going to see, but I think it's the most functional to read all the information and identifying every different thing. Anyway, I'm open minded, so I can heard your suggestions.

**Comment by Ashark** [comment]
No, I was not talking about blank values, but about blank headers. Here is what it looks like:
![blank headers](https://user-images.githubusercontent.com/22634975/123881607-beb25700-d94d-11eb-8786-150dfcb5ec6f.png)

**Comment by Ashark** [comment]
I am sorry, blank titles were because of incorrect translation file, which I saved when doing translation merge request. Ignore that statement.

> it isn't the most usable interface, but I think it's the most functional to read all the information and identifying every different thing. Anyway, I'm open minded, so I can heard your suggestions.

Yes, I see, kdialog cannot handle more complex gui.

I have seen another variant of dialog tool called yad: [Project homepage](https://github.com/v1cont/yad), [screenshots with complex guis](https://sanana.kiev.ua/index.php/yad). Unfortunately, it is on gtk.

Another solution is to use qt. I do not know yet if it is possible to create qt gui right from bash, but it can be done via pyqt or c++. Do not know if it worth to rewrite this project on c++, but pyqt seems good to me! I had done pyqt guis several times.

**Comment by Ashark** [comment]
Take a look at how ui could look:
![Screenshot_20210630_114739](https://user-images.githubusercontent.com/22634975/123934339-dd3f3f00-d99b-11eb-9c4d-7e01850ac224.png)

The file for this was made in qt designer, take it [here](https://github.com/Ashark/KDE-5-Template-Manager/blob/de937d60ce213c00edd52d7807fd5241dce7404e/untitled.ui).

**Comment by gmm96** [comment]
> I am sorry, blank titles were because of incorrect translation file, which I saved when doing translation merge request. Ignore that statement.
> 
> > it isn't the most usable interface, but I think it's the most functional to read all the information and identifying every different thing. Anyway, I'm open minded, so I can heard your suggestions.
> 
> Yes, I see, kdialog cannot handle more complex gui.
> 
> I have seen another variant of dialog tool called yad: [Project homepage](https://github.com/v1cont/yad), [screenshots with complex guis](https://sanana.kiev.ua/index.php/yad). Unfortunately, it is on gtk.
> 
> Another solution is to use qt. I do not know yet if it is possible to create qt gui right from bash, but it can be done via pyqt or c++. Do not know if it worth to rewrite this project on c++, but pyqt seems good to me! I had done pyqt guis several times.

Great! I was also thinking about that cause, as it has also happened to me before.
I was looking for dialog tools before starting the project and finally decided for kdialog because is in KDE base and there is no need to download additional packages or dependencies.

> Take a look at how ui could look:
> ![Screenshot_20210630_114739](https://user-images.githubusercontent.com/22634975/123934339-dd3f3f00-d99b-11eb-9c4d-7e01850ac224.png)
> 
> The file for this was made in qt designer, take it [here](https://github.com/Ashark/KDE-5-Template-Manager/blob/de937d60ce213c00edd52d7807fd5241dce7404e/untitled.ui).

Well, it looks pretty good, but, to be honest, I have never worked using qt, so I would need a further research.
I will release the current version with the editor and then we'll see.

> ![blank headers](https://user-images.githubusercontent.com/22634975/123881607-beb25700-d94d-11eb-8786-150dfcb5ec6f.png)

By the way, I can see in this screenshot Template Editor window icon (with value `edit`) is not showing for you.
Don't you have it? Can you run `kdialog --geticon` and search for `edit`?

**Comment by Ashark** [comment]
> I have never worked using qt. 

It is easier that you could think [I am talking about pyqt]. I also know nothing about it recently, but found a tutorial of how to work with it, and surprised how easy it was. It is in no way more complicated than bash!

> Can you run kdialog --geticon and search for edit?

I do not have an icon with value `edit`:
![Screenshot_20210702_055257](https://user-images.githubusercontent.com/22634975/124213975-2b6a5480-dafa-11eb-95c9-0383bb76b52b.png)

**Comment by gmm96** [comment]
> > I have never worked using qt.
> 
> It is easier that you could think [I am talking about pyqt]. I also know nothing about it recently, but found a tutorial of how to work with it, and surprised how easy it was. It is in no way more complicated than bash!
> 
> > Can you run kdialog --geticon and search for edit?
> 
> I do not have an icon with value `edit`:
> ![Screenshot_20210702_055257](https://user-images.githubusercontent.com/22634975/124213975-2b6a5480-dafa-11eb-95c9-0383bb76b52b.png)

Well, let's see if I have free time in the next vacations to take a look of it.

WTF, you have all the edit-whatever icons, but no singe edit icon. This shouldn't be serious. Maybe `editor` could be a good option for it.
Do you have `favorite` and `delete` icons? Just to be sure.

**Comment by Ashark** [comment]
Yes:
![favorite](https://user-images.githubusercontent.com/22634975/124367710-74d5b380-dc62-11eb-90ff-001b4985f9a3.png)
![delete](https://user-images.githubusercontent.com/22634975/124367711-769f7700-dc62-11eb-94cf-1cec58abaf25.png)

Note, there is no `xapp-favorite-symbolic` icon, as I said in #5

**Comment by Ashark** [comment]
Also, I have checked if such file should exist and belong to any package, got this:
```
$ pkgfile /usr/share/icons/breeze/actions/22/favorite.svg
extra/breeze-icons
$ pkgfile /usr/share/icons/breeze/actions/22/edit.svg
$
```

**Comment by Ashark** [comment]
> Maybe editor could be a good option for it.

Why not to just use the icon for the project? That is that icon png file with paper and letter T. Can you create an svg version and use it?

**Comment by gmm96** [comment]
> Yes:
> ![favorite](https://user-images.githubusercontent.com/22634975/124367710-74d5b380-dc62-11eb-90ff-001b4985f9a3.png)
> ![delete](https://user-images.githubusercontent.com/22634975/124367711-769f7700-dc62-11eb-94cf-1cec58abaf25.png)
> 
> Note, there is no `xapp-favorite-symbolic` icon, as I said in #5

> Also, I have checked if such file should exist and belong to any package, got this:
> 
> ```
> $ pkgfile /usr/share/icons/breeze/actions/22/favorite.svg
> extra/breeze-icons
> $ pkgfile /usr/share/icons/breeze/actions/22/edit.svg
> $
> ```

Thanks for the help. I will try to avoid `xapp-favorite-symbolic` icon, then.

> Why not to just use the icon for the project? That is that icon png file with paper and letter T. Can you create an svg version and use it?

I have tried to create different svg icons using Adobe Photoshop and the result was not fine. Depends on the KDE applied theme, the icon is displayed properly or not, specially using dark themes. I will use system default icons by now, until I would have a proper workaround to deal with it. You can check the icons I mentioned below.

[icons.zip](https://github.com/gmm96/KDE-5-Template-Manager/files/6808284/icons.zip)
